CMSgov/easi-app
# Easi-1645 --  Validation Error on Client Side for LCID Issuing and Extending

# EASI-1645

## Changes and Description

- Changes the validation for the date to include the following
  -  Month max number to be 12
  -  Day max number to be 31
  -  To verify that it is a valid date (i.e. 2/31/20xx will fail) 

|||
|--|:--:|
|All errors|![image](https://user-images.githubusercontent.com/2349702/159767652-22bfb68d-7bdc-46ea-b4bf-71ff211f386f.png)|
|Month error|![image](https://user-images.githubusercontent.com/2349702/159767827-88da471e-e586-4f6d-bce0-0d22c185bad2.png)![image](https://user-images.githubusercontent.com/2349702/159767931-46f05af0-0a29-45d1-b997-886f7700838e.png)|
|Day error|![image](https://user-images.githubusercontent.com/2349702/159768013-8109f31a-bfea-4cdb-87d1-126c347e2520.png)|
|Date error|![image](https://user-images.githubusercontent.com/2349702/159768098-6d71e30d-cd52-4d78-ac61-9b716215f9b8.png)|




## How to test this change

1. Navigate to a Intake Request
2. Click `Actions` in the sidebar
3. Click `Show other options`
4. Click `Issue Lifecycle Id`
5. Ensure the following dates throws error on client side
    1. mm/dd/yyyy
    2. 43/21/1234
    3. 12/34/1234
    4. 2/31/1234
6. Ensure that when there are no errors, the form gets submitted
7. Verify the above steps for Extending the LCID


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
